proto/agent.go

```diff
@@ -133,6 +131,11 @@ func (a *Agent) startNextTask(ctx context.Context, tc *taskContext) error {
 		"task_secret": tc.task.Secret,
 	})
 
+	// Defers are LIFO. We cancel all agent task threads, then any procs started by the agent, then remove the task directory.
+	defer a.removeTaskDirectory(tc)
```

Comment:
True, but we will not have created the task directory or any procs, and we will not have anything to cancel.